redaxo/src/addons/mediapool/lib/perm_read.php

```diff
@@ -0,0 +1,19 @@
+<?php
+
+/**
+ * @package redaxo\mediapool
+ */
+class rex_media_read_perm extends rex_media_perm
```

Comment:
Ist die Vererbung so sinnvoll? Also dass rex_media_read_perm dann instanceof rex_media_perm ist?

Ggf. nicht vererben, und die gemeinsamen Methoden in eine abstrakte Klasse, oder in ein Trait auslagern?

Reply:
War der weg des geringsten Wiederstandes. Kann man gerne noch hübschen, programmatisch ist es so jedenfalls verwendbar.

Reply:
Am liebsten wäre mir eigentlich, wir hätte nur eine media-perm-klasse, an der man dann alles abfragen kann (Schreibrechte, Leserechte..).

Aber dafür müsste man wohl die complex-perm-basisklasse im Core erweitern, da die aktuell ja immer nur eine ALL-Checkbox und eine Selectbox darstellen kann. 

Wobei ich es auch im UI eigentlich besser fände, wenn es irgendwie eine kombinierte Auswahl wäre. 
Aber wie das genau aussehen könnte, bin ich mir auch noch nicht sicher.
Ich denke da nochmal weiter drüber nach.

Reply:
<img width="874" alt="Bildschirmfoto 2020-06-07 um 22 57 32" src="https://user-images.githubusercontent.com/791247/83979977-5fcd0a80-a912-11ea-8a9f-0fdbba3901c6.png">

Etwas in diese Richtung wäre nett.

Reply:
Man kann die UI immer noch anpassen, aber fürs Erste funktioniert es ja auch so oder?

Reply:
> <img alt="Bildschirmfoto 2020-06-07 um 22 57 32" width="874" src="https://user-images.githubusercontent.com/791247/83979977-5fcd0a80-a912-11ea-8a9f-0fdbba3901c6.png">
> 
> Etwas in diese Richtung wäre nett.

Ich denke das geht wegen der Hierarchie nicht so gut so

Reply:
Denke ich nicht. Die Lösung aktuell macht es eigentlich nicht anders.

Reply:
Eine andere Alternative wäre es direkt am Baum zu machen und neben den Kategorien Checkboxen oder selects zur Auswahl anzeigen. Ich weiß nicht was da dann mehr Hausmeisterarbeit verursacht

Reply:
@gharlan was machen wir nun?
- Vererben wie gehabt
- core-Änderungen und alles in eine Klasse 
- 2 Klassen ohne Vererbung

Reply:
Schwierig finde ich. Ich tendiere zu folgender Lösung:

- Beide Klassen belassen, aber ohne Vererbung
- UI bleibt also so
- Abfragemethoden aber nur im `media`-Complexperm, also als Rechteabfrager hat man nur mit der einen Klasse zu tun, die `media_read` ist ein Implementierungsdetail
- Die `media`-Klasse holt sich intern die `media_read`-Rechte, und bietet dann Methoden für die kombinierte Auswertung

Das hätte auch den Vorteil, dass wir zukünftig ggf. doch mal den Core ändern können und das UI anpassen und alles in ein Element holen können. Wäre dann kein BC-Break, da die Abfragemethoden sowieso schon in der einen Klasse sich befänden.